### DISCUSSION TITLE
Flushing all caching keys on update or create

### DISCUSSION THREAD

**Discussion by alihassan161820** [directus/directus#13401] [Feature Requests]
I facing a problem while using caching that when ever i create or update any collection the whole caching key flushed 
I there is any way to set models names that when i updated them the cache flush

Example: if i has two models ( companies, reviews ) 

i need to flush cache only if i made updates in companies not in companies or reviews

### To Reproduce

Cache configuration:  

CACHE_AUTO_PURGE = true

**Comment by azrikahar** [comment]
> I there is any way to set models names that when i updated them the cache flush

There's no native support for this use case at the moment. With that said, you may be able to tweak [other cache configs](https://docs.directus.io/configuration/config-options/#cache) around such as `CACHE_SYSTEM_TTL`, depending on what's warranting you to configure `CACHE_AUTO_PURGE=true` in the first place 👍

This is also another rather random idea, but perhaps you can sort of achieve the selective-cache-flushing you were inquiring with a combination of action hooks + the cache clear utils endpoint: https://docs.directus.io/reference/system/utilities/#clear-the-internal-cache. Though I've never actually try this so unfortunately I can't confirm will this work exactly as your description.

**Comment by spawnrider** [comment]
Hi guys, 

We had the same issue / behaviour on our Directus instance. 
We are putting a lot of data (point of interests, countries, towns, ...) in cache (Redis instance). 
When an user is submitting a comment, the whole cache keys are deleted in Redis, this is taking approx. 6-7 seconds. 
An user comment shouldn't delete all cached data but only cache keys (i.e. answers) associated to this specific collection of data.

What could be done ?

**Reply by rijkvanzanten** [reply]
You could locally checkout pull request https://github.com/directus/directus/pull/15537, kick the tires, and leave a review! The more eyes on that PR, the faster it can get merged and released 🚀
